src/git/git.go

```diff
@@ -53,6 +72,345 @@ func (g *GitVersioning) validate() error {
 	return nil
 }
 
+func (g *GitVersioning) GetChangeHash() (string, error) {
+	if g.mostRecentCommit.hash == "" {
+		return "", errors.New("last commit hash not set")
+	}
+	return g.mostRecentCommit.hash, nil
+}
+
+func (g *GitVersioning) GetChangeAuthorName() (string, error) {
+	if g.mostRecentCommit.authorName == "" {
+		return "", errors.New("last commit name not set")
+	}
+	return g.mostRecentCommit.authorName, nil
+}
+
+func (g *GitVersioning) GetChangeAuthorEmail() (string, error) {
+	if g.mostRecentCommit.authorEmail == "" {
+		return "", errors.New("last commit email not set")
+	}
+	return g.mostRecentCommit.authorEmail, nil
+}
+
+func (g *GitVersioning) GetChangeMessage() (string, error) {
+	fmt.Println(g.mostRecentCommit.message)
+	if g.mostRecentCommit.message == "" {
+		return "", errors.New("last commit message not set")
+	}
+	return g.mostRecentCommit.message, nil
+}
+
+func (g *GitVersioning) GetCurrentVersion() (string, error) {
+	if g.mostRecentTag == "" {
+		return "", errors.New("most recent tag not set")
+	}
+	return g.mostRecentTag, nil
+}
+
+func (g *GitVersioning) UpgradeRemoteRepository(newVersion string) error {
+	// consider 1.0.0 as the start tag of a repository when it does not have tags yet
+	if newVersion == "0.1.0" || newVersion == "0.0.1" {
+		newVersion = "1.0.0"
+	}
+
+	if err := g.commitChanges(newVersion); err != nil {
+		return err
+	}
+
+	if err := g.push(); err != nil {
```

Comment:
what happens if some new version failed before finish all these steps? will we need some cleanup to try again? @esequielvirtuoso